coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -273,20 +278,63 @@ static String getFieldName(String oldName, String suggestedNewName) {
     return newName;
   }
 
-  static String generateDocumentationForLiteral(RexLiteral rexLiteral) {
+  private static String getLiteralValueAsString(@Nonnull RexLiteral rexLiteral) {
     StringWriter documentationWriter = new StringWriter();
     PrintWriter printWriter = new PrintWriter(documentationWriter);
 
     rexLiteral.printAsJava(printWriter);
     printWriter.flush();
 
-    return "Field created from view literal with value: " + documentationWriter;
+    return documentationWriter.toString();
   }
 
-  static String generateDocumentationForAggregate(AggregateCall aggregateCall) {
+  /**
+   * Given an input {@link RelNode} and the index of a field in the {@link RelNode}'s corresponding Avro schema,
+   * determine if the field with the specified index is a column from a table.
+   * @param fieldIndex the index of a field in the <code>inputRelNode</code>'s corresponding Avro schema
+   * @param inputRelNode the input {@link RelNode}
+   * @return true if the field at <code>fieldIndex</code> is a column from a table
+   */
+  private static boolean isColumn(int fieldIndex, @Nonnull RelNode inputRelNode) {
+    return !(inputRelNode instanceof LogicalAggregate)
+        || fieldIndex < ((LogicalAggregate) inputRelNode).getGroupSet().cardinality();
+  }
+
+  static String generateDocumentationForLiteral(@Nonnull RexLiteral rexLiteral) {
+    return "Field created from view literal with value: " + getLiteralValueAsString(rexLiteral);
+  }
+
+  static String generateDocumentationForAggregate(@Nonnull AggregateCall aggregateCall) {
     return "Field created in view by applying aggregate function of type: " + aggregateCall.getAggregation().getKind();
   }
 
+  static String generateDocumentationForFunctionCall(@Nonnull RexCall rexCall, @Nonnull Schema inputSchema,
+      @Nonnull RelNode inputRelNode) {
+    StringJoiner args = new StringJoiner(", ");
+
+    for (RexNode rexNode : rexCall.getOperands()) {
+      SqlKind nodeKind = rexNode.getKind();
+      switch (nodeKind) {
+        case LITERAL:
+          args.add(getLiteralValueAsString((RexLiteral) rexNode));
+          break;
+        case INPUT_REF:
+          int fieldIndex = ((RexInputRef) rexNode).getIndex();
+          if (isColumn(fieldIndex, inputRelNode)) {
```

Comment:
To provide more context: we only want to extract the field name if the field refers to a column, otherwise the name is not meaningful (they could be auto-assigned names such as `EXPR_3` or `_f0`).